psake.ps1

```diff
@@ -102,7 +102,7 @@ Task Build {
     }
 }
 
-Task Test -depends Build  {
+Task Test -depends Build {
```

Comment:
Running the "build" on Linux without Mono available meant that `dotnet461` wasn't available, so the build would fail, even though it isn't needed to run the tests when targeting `netcoreapp2.0`.  That was why building/testing in one step was the best way to success in all cases, because if `netcoreapp2.0` was available, then it would be able to build the dependencies.

I realize you modified the Travis CI scripts to include Mono, but it shouldn't be strictly necessary.

~_Sorry, put the previous comment on the wrong commit, so it made less sense._~ _Never-mind, it has to be on the most recent of that line change to be visible._

Reply:
If you want the CI tasks to build for all frameworks, my proposal would be to add `Build` to the `-TaskList` in each of the CI scripts.  If the project is already built, then `dotnet run` would have minimal work to do and should not need the `--no-build` option.

Separately in #1626, the command-line `ShowBuildMenu.bat` uses the psake Test to build/run all combinations of frameworks and bitness

Reply:
>If the project is already built, then dotnet run would have minimal build work to do and should not need the --no-build option.

In theory. But in real life it does build again.

You can check this by adding `-v n` to the run command.